src/java/htsjdk/samtools/cram/encoding/reader/RefSeqIdReader.java

```diff
@@ -0,0 +1,237 @@
+						readFeatures.add(readBase);
+						break;
+					case Substitution.operator:
+						final Substitution substitution = new Substitution();
+						substitution.setPosition(pos);
+						final byte code = baseSubstitutionCodec.readData();
+						substitution.setCode(code);
+						readFeatures.add(substitution);
+						break;
+					case Insertion.operator:
+						final Insertion insertion = new Insertion(pos, insertionCodec.readData());
+						readFeatures.add(insertion);
+						break;
+					case SoftClip.operator:
+						final SoftClip softClip = new SoftClip(pos, softClipCodec.readData());
+						readFeatures.add(softClip);
+						break;
+					case HardClip.operator:
+						final HardClip hardCLip = new HardClip(pos, hardClipCodec.readData());
+						readFeatures.add(hardCLip);
+						break;
+					case Padding.operator:
+						final Padding padding = new Padding(pos, paddingCodec.readData());
+						readFeatures.add(padding);
+						break;
+					case Deletion.operator:
+						final Deletion deletion = new Deletion(pos, deletionLengthCodec.readData());
+						readFeatures.add(deletion);
+						break;
+					case RefSkip.operator:
+						final RefSkip refSkip = new RefSkip(pos, refSkipCodec.readData());
+						readFeatures.add(refSkip);
+						break;
+					case InsertBase.operator:
+						final InsertBase insertBase = new InsertBase(pos, baseCodec.readData());
+						readFeatures.add(insertBase);
+						break;
+					case BaseQualityScore.operator:
+						final BaseQualityScore baseQualityScore = new BaseQualityScore(pos,
+								qualityScoreCodec.readData());
+						readFeatures.add(baseQualityScore);
+						break;
+					case Bases.operator:
+						final Bases bases = new Bases(pos, basesCodec.readData());
+						readFeatures.add(bases);
+						break;
+					case Scores.operator:
+						final Scores scores = new Scores(pos, scoresCodec.readData());
+						readFeatures.add(scores);
+						break;
+					default:
+						throw new RuntimeException("Unknown read feature operator: " + operator);
+					}
+				}
+
+				// mapping quality:
+				cramRecord.mappingQuality = mappingScoreCodec.readData();
+				if (cramRecord.isForcePreserveQualityScores()) {
+					cramRecord.qualityScores = qualityScoresCodec.readDataArray(cramRecord.readLength);
+				}
+			} else {
+				if (cramRecord.isUnknownBases()) {
+					cramRecord.readBases = SAMRecord.NULL_SEQUENCE;
+					cramRecord.qualityScores = SAMRecord.NULL_QUALS;
+				} else {
+					final byte[] bases = new byte[cramRecord.readLength];
+					for (int i = 0; i < bases.length; i++)
+						bases[i] = baseCodec.readData();
+					cramRecord.readBases = bases;
+
+					if (cramRecord.isForcePreserveQualityScores()) {
+						cramRecord.qualityScores = qualityScoresCodec.readDataArray(cramRecord.readLength);
+					}
+				}
+			}
+
+			recordCounter++;
+
+		} catch (final Exception e) {
+			System.err.printf("Failed at record %d. \n", recordCounter);
+			throw new RuntimeException(e);
+		}
```

Comment:
I'm not sure why we need such wide exception handling here (if there is a specific case that can happen, maybe a narrower exception would be good) but we should at least change the printf.